### PR TITLE
Update javascript a litle bit 

### PR DESCRIPTION
**Pull request by Fedik**
Check, what you think? :smile:

### CODE REVIEW

**Comment by Bakual** [comment]
Looks better than what I wrote for sure. I just don’t have yet a clue what you did beside moving it to an own file. 
Will test tomorrow!

From: Fedir Zinchuk 
Sent: Tuesday, August 11, 2015 7:48 PM
To: Bakual/joomla-cms 
Subject: [joomla-cms] Update javascript a litle bit (#5)

Check, what you think? 

---

You can view, comment on, or merge this pull request online at:
  https://github.com/Bakual/joomla-cms/pull/5

Commit Summary
  a.. Update javascript a litle bit :smile: 
  b.. Merged by accident 
File Changes
  a.. M administrator/modules/mod_sampledata/mod_sampledata.xml (3) 
  b.. M administrator/modules/mod_sampledata/tmpl/default.php (57) 
  c.. A media/mod_sampledata/js/sampledata-process.js (91) 
Patch Links:
  a.. https://github.com/Bakual/joomla-cms/pull/5.patch 
  b.. https://github.com/Bakual/joomla-cms/pull/5.diff

**Comment by Fedik** [comment]
sorry for delay,

yes I just move to js file, and did some related changes to make it work from file :smile: 
and did small markup changes, now progress is hidden, and will be displayed only when something in progress ..
also as `response.data` can contain messages from couple plugins, I added render each message separately

also there is check for avoid execute the code multiple time at once

File: media/mod_sampledata/js/sampledata-process.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    	request.fail(function(jqXHR, textStatus){
+    		alert('Something went wrong! Please reboot the Windows and try again!');
+    	});
+	};
+
+	window.sampledataApply = function(el) {
+		var $el = $(el), type = $el.data('type'), steps = $el.data('steps');
+
+		// Check whether the work in progress or we alredy proccessed with current item
+		if (inProgress || $el.data('processed')) {
```
Comment (Fedik) [inline media/mod_sampledata/js/sampledata-process.js]: this prevent execution if something already in progress, or this item already was executed

File: media/mod_sampledata/js/sampledata-process.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+			return;
+		}
+
+		// Make sure that use run this not by random clicking on the page links
+		if (!confirm(Joomla.JText._('MOD_SAMPLEDATA_CONFIRM_START'))) {
```
Comment (Fedik) [inline media/mod_sampledata/js/sampledata-process.js:83]: make sure that user do not clicked the link accidentally :smile:

File: media/mod_sampledata/js/sampledata-process.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    		}
+		});
+		request.done(function(response){
+			$stepLi.children('img').remove();
+
+			if (response.success && response.data && response.data.length > 0) {
+				var success, value, resultClass, $msg;
+
+				// Display all messages that we got
+				for(var i = 0, l = response.data.length; i < l; i++) {
```
Comment (Fedik) [inline media/mod_sampledata/js/sampledata-process.js:40]: render all messages from `response.data`

File: media/mod_sampledata/js/sampledata-process.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+			$stepLi = $('<li class="' + stepClass + '"><img src="' + window.modSampledataIconProgress + '" width="30" height="30" ></li>'),
+			$progress = $(".sampledata-progress-" + type + " progress");
+
+		$("div.sampledata-progress-" + type + " ul").append($stepLi);
+
+		var request = $.ajax({
+			url: window.modSampledataUrl,
+    		type: 'POST',
+    		dataType: 'json',
+    		data: {
```
Comment (Fedik) [inline media/mod_sampledata/js/sampledata-process.js]: simpler way to add the request query :wink:

File: media/mod_sampledata/js/sampledata-process.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+		}
+
+		// Make sure that use run this not by random clicking on the page links
+		if (!confirm(Joomla.JText._('MOD_SAMPLEDATA_CONFIRM_START'))) {
+			return false;
+		}
+
+		// Turn on the progress container
+		$('.sampledata-progress-' + type).show();
+		$el.data('processed', true)
```
Comment (Fedik) [inline media/mod_sampledata/js/sampledata-process.js:89]: set marker that element is processed...
also if add `data-processed="true"` (when render the markup in PHP) to the link element then the script will see it and prevent from execution it second time

**Comment by Bakual** [comment]
@Fedik Thanks for the explanations. Sounds great :+1: 
Sorry for not having it tested yet. I had a few other tasks which came between.

**Comment by Fedik** [comment]
really no problem, I am not hurry :smile:

**Comment by Bakual** [comment]
@Fedik I played around a bit today. Looks great, I like it :+1: 
Some things I noticed:

### Progress Bar

If there is an error like there is in the blog sampledata one (more steps defined than present), there is an empty respons. This error is catched from the module and shows correct but the progress bar disappears. I think the bar should stay like in other error conditions.
missing progress
There is also a design issue with the progress bar when it is shown (using Chrome):
bar

### Execution Prevention

It works, but a bit to good I think. I would only restrict installing the same sampledata twice. Currently if you for whatever reason want to install another one, it will be prevented.
Also, there is no message. It just does nothing at all. I think it would need at least an alert box telling the user to not click the sampledata twice. Otherwise it just feels broken.

### Progress Icon

I think it looks better when it is centered like it was before your change.
icon

**Comment by Fedik** [comment]
thanks for review :wink: 

> the progress bar disappears

Here is I was not sure what to do, in old behavior there was `.val(0)`, and I thought more simple just hide it :smile: 
but I think you are right, better display it as is...
Progress bar styling I will check also, I think there I missed something in markup...

> Currently if you for whatever reason want to install another one, it will be prevented.

I tried make that in one time can be executed only one Item, 
means you cannot start "link 1"  and while it in progress also run "link 5"
but add some message is good idea 

> Progress Icon

ok, I will change it back :smile:

**Comment by Fedik** [comment]
ok, changed that things :smiley:

**Comment by Bakual** [comment]
Tested and goign to merge now. Looks great thanks :+1:

**Pull request closed without merging**